test/static-thread-pool.cc

```diff
@@ -151,3 +163,27 @@ TEST(StaticThreadPoolTest, Spawn) {
 
   *e();
 }
+
+TEST(StaticThreadPoolTest, Concurrent) {
+  std::vector<int> digits = {1, 2, 3};
+
+  auto id = std::this_thread::get_id();
+  auto e = [&]() {
+    return StaticThreadPool::Spawn(
+        "concurrent test",
+        Iterate(digits)
+            | ConcurrentOrdered([&]() {
+                auto* context = Scheduler::Context::Get();
+                return Map(Let([&](int& i) {
+                  EXPECT_NE(id, std::this_thread::get_id());
```

Comment:
This line isn't really testing anything related to what you've introduced in this PR because you haven't specified any `requirements`. A better thing to test would be to have a requirement that you are pinned to a specific thread and then make sure that both the parent and child context are on the same thread!